libs/langgraph/langgraph/func/__init__.py

```diff
@@ -266,7 +279,35 @@ def my_workflow(input_data: str, previous: Optional[str] = None) -> str:
         ```
     """
 
-    def _imp(func: types.FunctionType) -> Pregel:
+    def __init__(
+        self,
+        checkpointer: Optional[BaseCheckpointSaver] = None,
+        store: Optional[BaseStore] = None,
+        config_schema: Optional[type[Any]] = None,
+    ) -> None:
+        """Initialize the entrypoint decorator."""
+        self.checkpointer = checkpointer
+        self.store = store
+        self.config_schema = config_schema
+
+    class final(Generic[R, S]):
+        """Return a value and save the state associated with the entrypoint."""
+
+        __slots__ = ("value", "save")
+
+        value: R
+        """Value to return. A value will always be returned even if it is None."""
+        save: S
+        """The value for the state for the next checkpoint. 
+
+        A value will always be saved even if it is None.
+        """
+
+        def __init__(self, value: R, save: S) -> None:
```

Comment:
@nfcampos thoughts on making these named arguments only given that `entrypoint.final` is not as self-explanatory? it's a lot to type `entrypoint.final(value='hello', save='goodbye')`

Reply:
Yea I agree with making them kwargs